sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -207,11 +207,19 @@ private[sql] object ParquetFilters {
      */
   }
 
+  private def getFieldMap(dataType: DataType): Array[(String, DataType)] = dataType match {
```

Comment:
It's not quite intuitive why this method is needed. Can we add a comment here to explain why fields with "optional" metadata should be filtered? Corresponding JIRA links should also be mentioned.